policy.go

```diff
@@ -43,7 +44,7 @@ const (
 // Kubernetes pods local traffic. The def is a policy to use when an upstream
 // address doesn't match the skipHeaderCIDR.
 func SkipProxyHeaderForCIDR(skipHeaderCIDR *net.IPNet, def Policy) PolicyFunc {
-	return func(upstream net.Addr) (Policy, error) {
+	return func(upstream net.Addr, downstream net.Addr) (Policy, error) {
```

Comment:
This breaks the API and I'm thinking there's no need to do this. `upstream` is just a placeholder for any `net.Addr` so unless both upstream and downstream addresses have to be evaluated in a policy, I think we can just keep the API as is, maybe rename `upstream` to `address` and the new `PolicyFunc` just works. WDYT?

Reply:
I thought about it and the issue i faced is where this function is called https://github.com/pires/go-proxyproto/blob/main/protocol.go#L71. If we have 2 policy function one calling with the remote and other with the local address, then i felt its not straight forward on handling different policy returned when both the functions are used.
I can change it if we are okay with second function overriding if both are set.

Reply:
@pires what do you suggest as i don't see any other option apart from these?

Reply:
I'm thinking if anything, we'd split `PolicyFunc` into `DownstreamPolicyFunc` and `UpstreamPolicyFunc` with the same signature. My only _strong_ opinion is not wanting to break the API.

Reply:
@emersion got any thoughts on this? 🙏🏻

Reply:
Yes, i understand that. Do we want to have one function override the other if both are present or do we want to pick a stricter policy of both(which i think is bit difficult to say which is stricter)